xlcalculator/xlfunctions/utils.py

```diff
@@ -15,3 +15,7 @@ def datetime_to_number(value):
     # Excel treats 1900 as a leap year.
     offset = 2 if delta.days > 58 else 1
     return (delta.days + offset) + (delta.seconds / 24 * 60 * 60)
+
+
+def XlDateTime_to_datetime(value):
+    return number_to_datetime(int(value))
```

Comment:
Note that XlDateTime does not need to be an integer. All decimal places specify the time. And once you have that, what's the point of the function?

Reply:
that is currently only used in XIRR. There's a potential it doesn't need to be. For a while I was struggling in XNPV with the date part of the equation (date - min_date). I'll reconsider and give that another go.

From memory I also used this operation in EDATE and EOMONTH and thought it might have been handy to outsource it to a util method.

Reply:
Have removed this method.

I still need to cast dates and guess to make newton optimization happy, but have done that in the XIRR function itself.

Reply:
I think that's better.